### PR TITLE
Implement NSArray's customMirror

### PR DESCRIPTION
**Pull request by noppefoxwolf**

### CODE REVIEW

**Comment by ikesyo** [comment]
Let's kicking CI.

@swift-ci Please test.

**Comment by ikesyo** [comment]
`swift-4.2-branch` is limited to pull requests for bug fixes which are already landed on `master`. Could you target this  on `master`?

See https://forums.swift.org/t/announcing-swift-4-2-2-and-monthly-swift-4-2-x-dot-releases-for-linux/20148 for the details.

**Comment by ikesyo** [comment]
@noppefoxwolf `swift-4.2-branch` is limited to pull requests for bug fixes which are already landed on `master`. Could you target this  on `master`?

See https://forums.swift.org/t/announcing-swift-4-2-2-and-monthly-swift-4-2-x-dot-releases-for-linux/20148 for the details.

**Comment by noppefoxwolf** [comment]
@ikesyo Thank you for your comment!
I changed target and resolved conflict.

File: Foundation/NSArray.swift [hunk 1]
```diff
@@ -952,7 +952,9 @@ extension NSArray : ExpressibleByArrayLiteral {
 }
 
 extension NSArray : CustomReflectable {
-    public var customMirror: Mirror { NSUnimplemented() }
+    public var customMirror: Mirror {
+        return Mirror(reflecting: Array(self))
```
Comment (ikesyo) [inline Foundation/NSArray.swift]: How about using `_storage`?

```swift
return Mirror(reflecting: _storage)
```

File: TestFoundation/TestNSArray.swift [hunk 2]
```diff
@@ -796,6 +797,21 @@ class TestNSArray : XCTestCase {
         let match5 = paths.pathsMatchingExtensions(["..txt"])
         XCTAssertEqual(match5, [])
     }
+    
+    func test_customMirror() {
+        let inputArray = ["this", "is", "a", "test", "of", "custom", "mirror"]
+        let array = NSArray(array: inputArray)
+        let arrayMirror = array.customMirror
+        dump(arrayMirror.children)
```
Comment (ikesyo) [inline TestFoundation/TestNSArray.swift]: Is there any reason to leave this?

**Comment by ikesyo** [comment]
@swift-ci Please test

**Comment by noppefoxwolf** [comment]
@ikesyo Thank you!
I resolve comments issues.

**Comment by ikesyo** [comment]
@swift-ci Please test

**Comment by spevans** [comment]
@swift-ci Please test

**Comment by noppefoxwolf** [comment]
@ikesyo @spevans Thank you for checking. 
Is there anything else to do?

**Comment by ikesyo** [comment]
@swift-ci test and merge

**Comment by ikesyo** [comment]
@swift-ci test and merge

**Pull request merged**